### PR TITLE
build: Enable unit tests and integration tests to be run separately

### PR DESCRIPTION
**Pull request by gkc**
**- What I did**
Enable unit tests and integration tests to be run separately. Unit tests are under the src/test directory and will be run first, by the surefire plugin. Integration tests will be run if the unit tests pass; integration tests are in the src/integration-test directory and are run by the failsafe plugin. Integration tests are just JUnit test cases but they have "IT" at the start or the end of their names

**- How I did it**
* Added various maven incantations to pom.xml
* Added src/integration-test directory
* Added a do-nothing "integration test" to verify that (a) the test case runs (b) it runs in the "verify" phase and not the "test" phase
* Modified github action so the test-and-build job now does three separate mvn runs: (i) run unit tests (ii) run integration tests (iii) run install phase

**- How to verify it**
Tests pass

**- Description for the changelog**
build: Enable unit tests and integration tests to be run separately

### CODE REVIEW

**Comment by gkc** [comment]
@JeremyTubongbanua please take a look and double-check that I haven't broken any of the maven packaging work you did

File: .github/workflows/at_client.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ jobs:
           distribution: 'adopt'
           cache: maven
           server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
-
-      - name: Test and Build with Maven
-        run: mvn --batch-mode --update-snapshots --no-transfer-progress -Dgpg.skip=true clean install
+      - name: Build clean and run unit tests
+        run: mvn --batch-mode clean test -DskipIntegrationTests=true
```
Comment (JeremyTubongbanua) [inline .github/workflows/at_client.yml]: `-Dgpg.skip=true` add this flag too - I am scared `test` will run the signing plugin

**Review by JeremyTubongbanua — CHANGES_REQUESTED** [review]

**Comment by JeremyTubongbanua** [comment]
Note for me - I'll implement this [solution](https://stackoverflow.com/a/14869692) so that we can only perform a release when we say so.

File: .github/workflows/at_client.yml (hunk 1 above)
Comment (gkc) [reply .github/workflows/at_client.yml]: Done

**Review by JeremyTubongbanua — APPROVED** [review]

**Pull request merged**